nodejs/awsx/tests/ec2/ec2.spec.ts

```diff
@@ -139,324 +170,519 @@ describe("cidr", () => {
 
 describe("topology", () => {
     describe("default", () => {
-        it("1 AZ", () => {
-            assert.equal(topologyToJson("10.0.0.0/16", 1, [
+        it("1 AZ", async () => {
+            await subnetsEqual(subnets("10.0.0.0/16", oneAZ, [
                 { type: "public" },
                 { type: "private" },
-            ]), `[
+            ]), [
     {
         "type": "public",
         "subnetName": "testing-public-0",
-        "availabilityZone": 0,
-        "cidrBlock": "10.0.0.0/17"
+        args: {
+            "availabilityZone": AZ1.name,
+            "availabilityZoneId": AZ1.id,
+            "cidrBlock": "10.0.0.0/17",
+            mapPublicIpOnLaunch: true,
+            assignIpv6AddressOnCreation: false,
+        },

```

Comment:
none of the existing tests actually changed in terms of existing data they returned.  but they now return much more information than before (effectively, all the args necessary to create the child subnets).  i hand validated all of these.